addons/payment_stripe/models/payment.py

```diff
@@ -109,13 +109,15 @@ def _add_available_payment_method_types(self, stripe_session_data, tx_values):
         for idx, payment_method_type in enumerate(available_payment_method_types):
             stripe_session_data[f'payment_method_types[{idx}]'] = payment_method_type
 
-    def _stripe_request(self, url, data=False, method='POST'):
+    def _stripe_request(self, url, data=False, method='POST', idempotency_key=None):
         self.ensure_one()
         url = urls.url_join(self._get_stripe_api_url(), url)
         headers = {
             'AUTHORIZATION': 'Bearer %s' % self.sudo().stripe_secret_key,
-            'Stripe-Version': '2019-05-16', # SetupIntent need a specific version
-            }
+            'Stripe-Version': '2019-05-16',  # SetupIntent need a specific version
+        }
+        if method == 'POST' and (url.endswith('payment_intents') or url.endswith('refunds')) and idempotency_key:
```

Comment:
I'd rather let the calling method decide if we need the idempotency key for a given URL.
```suggestion
        if method == 'POST' and idempotency_key:
```